build/gen_stub.php

```diff
@@ -4029,25 +4035,23 @@ function generateFunctionEntries(?Name $className, array $funcInfos, ?string $co
 /**
  * @param iterable<FuncInfo> $funcInfos
  */
-function generateAttributeInitialization(iterable $funcInfos, ?string $parentCond = null): string {
+function generateAttributeInitialization(iterable $funcInfos, ?string $parentCond = null, iterable $allConstInfos): string {
     return generateCodeWithConditions(
         $funcInfos,
         "",
-        static function (FuncInfo $funcInfo) {
+        static function (FuncInfo $funcInfo) use ($allConstInfos) {
             $code = null;
 
             foreach ($funcInfo->args as $index => $arg) {
-                if (!$arg->isSensitive) {
-                    continue;
-                }
-
                 if ($funcInfo->name instanceof MethodName) {
                     $functionTable = "&class_entry->function_table";
                 } else {
                     $functionTable = "CG(function_table)";
                 }
 
-                $code .= "\tzend_mark_function_parameter_as_sensitive($functionTable, \"" . $funcInfo->name->getNameForAttributes() . "\", $index);\n";
+                foreach ($arg->attributes as $attribute) {
+                    $code .= $attribute->generateCode("zend_add_parameter_attribute(zend_hash_str_find_ptr($functionTable, \"" . $funcInfo->name->getNameForAttributes() . "\", sizeof(\"" . $funcInfo->name->getNameForAttributes() . "\") - 1), $index", "{$funcInfo->getArgInfoName()}_arg{$index}", $allConstInfos);
```

Comment:
I don't really like the fact that `$funcInfo->getArgInfoName()` is used for generating the attribute name, because the `arginfo_` part is unnecessary (and I didn't realize for a while why it's there). Even though the format of `$funcInfo->name->getMethodSynopsisFilename()` looks a bit better, it would be weird to use this name here.. So either we should rename this method (`getSnakeCasedName`??) or just leave the name as-is.

Reply:
Thanks! `$funcInfo->name->getMethodSynopsisFilename()` is indeed much better. It's was hard for me to "discover" something appropriate to use and `->getArgInfoName()` worked.